test/regression/builder/builder_tools_test.cc

```diff
@@ -189,29 +194,334 @@ TEST_F(SolveLaneEndsInnerLaneSectionTest, TestMethod) {
+// @param left_lane_user_data The userData XODR node value.
+// @param left_lane_direction The direction value for the left lane.
+// @param right_lane_user_data The userData XODR node value.
+// @param right_lane_direction The direction value for the right lane.
+// @returns A string that contains a XML description of a XODR Lane node.
+std::string GetXodrWithDirectingLanes(const std::string& road_rule = "RHT", const std::string& left_lane_user_data = "",
+                                      const std::string& left_lane_direction = "standard",
+                                      const std::string& right_lane_user_data = "",
+                                      const std::string& right_lane_direction = "standard") {
+  std::stringstream ss;
+  ss << "<?xml version='1.0' standalone='yes'?>" << std::endl;
+  ss << "<OpenDRIVE>" << std::endl;
+  ss << " <header revMajor='1' revMinor='1' name='Test' version='1.21' date='Tue Jul 1 12:00:00 2025' "
+     << "north='0' south='0' east='0' west='0' vendor='Toyota Research Institute' >" << std::endl;
+  ss << " </header>" << std::endl;
+  ss << " <road name='Road 0' length='1.0e+2' id='0' junction='-1' rule='" << road_rule << "'>" << std::endl;
+  ss << "   <planView>" << std::endl;
+  ss << "     <geometry s='0.0' x='0.0' y='0.0' hdg='1.' length='1.0e+2'>" << std::endl;
+  ss << "       <line/>" << std::endl;
+  ss << "     </geometry>" << std::endl;
+  ss << "   </planView>" << std::endl;
+  ss << "   <lanes>" << std::endl;
+  ss << "     <laneSection s='0.'>" << std::endl;
+  ss << "         <left>" << std::endl;
+  ss << "           <lane id='1' type='driving' direction='"
+     << (left_lane_direction.empty() ? "standard" : left_lane_direction) << "'>" << std::endl;
+  ss << "             <width sOffset='0.' a='2.0' b='0.' c='0.' d='0.'/>" << std::endl;
+  ss << "             <speed sOffset='0.1' max='45.' unit='mph'/>" << std::endl;
+  ss << "             " << left_lane_user_data << std::endl;
+  ss << "           </lane>" << std::endl;
+  ss << "         </left>" << std::endl;
+  ss << "         <center>" << std::endl;
+  ss << "           <lane id='0' type='none' />" << std::endl;
+  ss << "         </center>" << std::endl;
+  ss << "         <right>" << std::endl;
+  ss << "           <lane id='-1' type='driving' direction='"
+     << (right_lane_direction.empty() ? "standard" : left_lane_direction) << "'>" << std::endl;
+  ss << "             <width sOffset='0.' a='2.0' b='0.' c='0.' d='0.'/>" << std::endl;
+  ss << "             <speed sOffset='0.1' max='45.' unit='mph'/>" << std::endl;
+  ss << "             " << right_lane_user_data << std::endl;
+  ss << "           </lane>" << std::endl;
+  ss << "         </right>" << std::endl;
+  ss << "       </laneSection>" << std::endl;
+  ss << "     </lanes>" << std::endl;
+  ss << "   </road>" << std::endl;
+  ss << "</OpenDRIVE>" << std::endl;
+  return ss.str();
+}
+
+class LaneTravelDirTest : public ::testing::Test {
```

Comment:
nit(if time): Convert the following tests to TEST_P

Reply:
Not done. We can add it to the backlog.